sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/SqliteCompiler.kt

```diff
@@ -46,6 +47,7 @@ class SqliteCompiler {
       projectPath: String
   ): Status {
     try {
+      nameAllocators.put(parseContext.start.inputStream, NameAllocator())
```

Comment:
It looks like this can just be passed around. Or otherwise we can create an instance-based writer instead of one that basically holds no state.